### PR TITLE
Invite Flow (2/3): nick/mailer

### PR DESCRIPTION
**Pull request by niwong**
## What's new in this PR?

**BACKEND INVITE FLOW**

This PR does three things:
    1. Receive a pledge invitation's RECORD_ID generated from AirTable on the frontend via the `/invite` POST route.
    2. Retrieve that record given the RECORD_ID and extract the invitation recipient's email.
    3. Send an email using nodemailer to the extracted email CONTAINING a link to onboarding carrying the RECORD_ID.
  
### How to Review

Heads up, you can use your very own gmail. I've wrote it to only work with GMAIL. Here's what you need to do.

1. On lines 33-35, you must specify your GMAIL email, password, and name that you want the email to show up as.

2. Next, you must specify to Google that you're allowing your GMAIL to be used by "less secure apps." Navigate here, read the second paragraph, and click the link in the second paragraph that gives google the "OK" to let our backend use your email.
https://nodemailer.com/usage/using-gmail/. If you have 2FA on that boi, there are additional steps for you to take. I recommend using a email you don't really care about.

3. Pull my two branches on `peoplepower-web`, 1. `nick/admin-modal` AND 2. `nick/integrate-backend` to get the form that will trigger this email stuff to work.

### Related PRs

`nick/admin-modal` AND 2. `nick/integrate-backend` IN PEOPLEPOWER-WEB

### Next steps

[//]: # "What doesn't work yet, what's NOT in this PR/has to be done "

### Tests Performed, Edge Cases

[//]: # 'If you made changes/additions to `airtable.js` make sure you update/add'
[//]: # '[at least 1] corresponding unit tests in `airtable.spec.js`'

### Screenshots

[//]: # "Add screenshots!!! If you'd like"

CC: @dfangshuo

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -27,4 +29,43 @@ app.get('/', (req, res) => {
...
+
+  const baseUrl = 'https://peoplepower.netlify.com/onboarding';
+  const inviteParameter = `?token=${RECORD_ID}`;
+  const inviteLink = baseUrl + inviteParameter;
+
+  const info = await transporter.sendMail({
+    from: `"${SENDER_NAME}" <${MAIL_SERVER_EMAIL}>`, // sender address
+    to: recipientEmail, // list of receivers
+    subject: 'PP POWER invites you!!', // Subject line
+    text: `Welcome to People Power Solar Cooperative! To join, you can create your account via this link: ${inviteLink}`, // plain text body
```
Comment (aivantg) [inline index.js]: so this "text" and subject stuff could be exported to a file called like `EmailTemplates.js` which has two objects that looks something like this: 

```
export const subjects = {
  invite: () => (`PP POWER invites you!!'}
}

export const emailText = {
  invite: (inviteLink) => `Welcome to People Power Solar Cooperative! To join, you can create your account via this link: ${inviteLink}`
}

```

Comment (aivantg) [reply index.js]: you could a similar HTML object!

File: index.js [hunk 2]
```diff
@@ -27,4 +29,43 @@ app.get('/', (req, res) => {
...
+      user: MAIL_SERVER_EMAIL,
+      pass: MAIL_SERVER_PASS
+    }
+  });
+
+  const baseUrl = 'https://peoplepower.netlify.com/onboarding';
+  const inviteParameter = `?token=${RECORD_ID}`;
+  const inviteLink = baseUrl + inviteParameter;
+
+  const info = await transporter.sendMail({
```
Comment (aivantg) [inline index.js]: generalize this function!

File: index.js [hunk 3]
```diff
@@ -27,4 +29,43 @@ app.get('/', (req, res) => {
...
 });
 
-app.listen(port, () => console.log(`Express App Listening on port ${port}!`));
+app.post('/invite', async (req, res) => {
+  const MAIL_SERVER_EMAIL = ''; // this is the GMAIL you want to send emails from. "nick@gmail.com"
+  const MAIL_SERVER_PASS = ''; // this is your GMAIL password
+  const SENDER_NAME = ''; // specify the name of the person sending an email, for example: "Nick Wong ⚡️🔋"
+
+  const RECORD_ID = req.body.pledgeInviteID;
+  const pledgeInvite = await getPledgeInviteById(RECORD_ID);
```
Comment (aivantg) [inline index.js]: probably wanna verify that this returned a record

File: .env.example [hunk 4]
```diff
@@ -1,3 +1,5 @@
 ENPHASE_KEY=
-REACT_APP_AIRTABLE_API_KEY=
-UTILITY_API_KEY=
+SAMPLE_ENPHASE_USER_ID=
```
[the remark this answers is not in the document]
Comment (niwong) [reply .env.example]: Checked-out the bill gen PR, but saw that `REACT_APP_AIRTABLE_API_KEY` was added in there.  Which PR should I follow?

File: index.js [hunk 5]
```diff
@@ -27,4 +28,19 @@ app.get('/', (req, res) => {
...
 });
 
-app.listen(port, () => console.log(`Express App Listening on port ${port}!`));
+app.post('/invite', async (req, res) => {
+  const RECORD_ID = req.body.pledgeInviteID;
+  const confirmSend = await sendInviteEmail(RECORD_ID);
+
+  if (confirmSend === '') {
+    res.send({
+      status: `An error occured when sending an invitation.`
```
Comment (aivantg) [inline index.js]: by default, I think `res.send` sends a response with a `200` code, which is generally consider okay. But since there was an error, we should send a something like a `400` code. In Express you can do this by doing something like: 
`res.status(400).send({status: "Error ocurred"})`

Comment (aivantg) [reply index.js]: Honestly, you don't need to change it, just a note for your info honestly lol. We're not using status codes properly anywhere else either so maybe when there's an error handling PR we'll be more meticulous. More on status codes here: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_errors

File: utils/email.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  host: 'smtp.gmail.com',
+  port: 587,
+  secure: false, // true for 465, false for other ports
+  auth: {
+    user: process.env.MAIL_SERVER_EMAIL,
+    pass: process.env.MAIL_SERVER_PASS
+  }
+});
+
+const retrieveRecipientEmail = async RECORD_ID => {
```
Comment (aivantg) [inline utils/email.js]: nice function organization!
